wasm/test_page/js/worker.js

```diff
@@ -0,0 +1,294 @@
+  // English as a pivot language.
+  if (from !== 'en' && to !== 'en') {
+    log(`Translating '${from}${to}' via pivoting: '${from}en' -> 'en${to}'`);
+    let translatedParagraphsInEnglish = translateInvolvingEnglish(from, 'en', paragraphs);
+    return translateInvolvingEnglish('en', to, translatedParagraphsInEnglish);
+  }
+  else {
+    log(`Translating '${from}${to}'`);
+    return translateInvolvingEnglish(from, to, paragraphs);
+  }
+}
+
+const translateInvolvingEnglish = (from, to, paragraphs) => {
+  const languagePair = `${from}${to}`;
+  if (!translationModels.has(languagePair)) {
+    throw Error(`Please load translation model '${languagePair}' before translating`);
+  }
+  translationModel = translationModels.get(languagePair);
+
+  // Instantiate the arguments of translate() API i.e. ResponseOptions and input (vector<string>)
+  var responseOptions = new Module.ResponseOptions();
+  let input = new Module.VectorString;
+
+  // Initialize the input
+  paragraphs.forEach(paragraph => {
+    // prevent empty paragraph - it breaks the translation
+    if (paragraph.trim() === "") {
+      return;
+    }
+    input.push_back(paragraph.trim())
+  })
+
+  // Access input (just for debugging)
+  log(`Input size: ${input.size()}`);
+
+  // Translate the input, which is a vector<String>; the result is a vector<Response>
+  let result = translationService.translate(translationModel, input, responseOptions);
+
+  const translatedParagraphs = [];
+  const translatedSentencesOfParagraphs = [];
+  const sourceSentencesOfParagraphs = [];
+  for (let i = 0; i < result.size(); i++) {
+    translatedParagraphs.push(result.get(i).getTranslatedText());
+    translatedSentencesOfParagraphs.push(getAllTranslatedSentencesOfParagraph(result.get(i)));
+    sourceSentencesOfParagraphs.push(getAllSourceSentencesOfParagraph(result.get(i)));
+  }
+
+  responseOptions.delete();
+  input.delete();
+  return translatedParagraphs;
+}
+
+// This function extracts all the translated sentences from the Response and returns them.
+const getAllTranslatedSentencesOfParagraph = (response) => {
+  const sentences = [];
+  const text = response.getTranslatedText();
+  for (let sentenceIndex = 0; sentenceIndex < response.size(); sentenceIndex++) {
+    const utf8SentenceByteRange = response.getTranslatedSentence(sentenceIndex);
+    sentences.push(_getSentenceFromByteRange(text, utf8SentenceByteRange));
+  }
+  return sentences;
+}
+
+// This function extracts all the source sentences from the Response and returns them.
+const getAllSourceSentencesOfParagraph = (response) => {
+  const sentences = [];
+  const text = response.getOriginalText();
+  for (let sentenceIndex = 0; sentenceIndex < response.size(); sentenceIndex++) {
+    const utf8SentenceByteRange = response.getSourceSentence(sentenceIndex);
+    sentences.push(_getSentenceFromByteRange(text, utf8SentenceByteRange));
+  }
+  return sentences;
+}
```

Comment:
```js

getAllTranslatedSentencesOfParagraph -> response.getTranslatedSentence(sentenceIndex)
getAllSourceSentencesOfParagraph -> response.getSourceSentence(sentenceIndex)

getSentences(response.target)
getSentences(response.source)

// Or just export annotation, wrap it nicely. 

```

Let's make it a point to call out how bad design is creating more bloat further down, making more work for everyone.